### PR TITLE
add scan test

### PR DESCRIPTION
**Pull request by morefreeze**
@qiuyesuifeng @disksing PTAL

### CODE REVIEW

File: scan_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+type testScanSuite struct {
+	store   *tikvStore
+	prefix  string
+	rowNums []int
+}
+
+var _ = Suite(&testScanSuite{})
+
+func (s *testScanSuite) SetUpSuite(c *C) {
+	d := Driver{}
```
Comment (ngaut) [inline scan_test.go]: var d Driver

File: scan_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	store, err := d.Open(fmt.Sprintf("tikv://%s%s?cluster=%d", *etcdAddrs, *pdLeaderPath, *clusterID))
+	c.Assert(err, IsNil)
+	s.store = store.(*tikvStore)
+	s.prefix = "seek_cd7540"
```
Comment (ngaut) [inline scan_test.go]: What does seek_cd7540 mean ?

Comment (morefreeze) [inline scan_test.go]: It's just a unique prefix.

Comment (morefreeze) [inline scan_test.go]: Or just `seek` is ok?

Comment (morefreeze) [inline scan_test.go]: It prevents from conflicting some key like `seek123` in store.

File: scan_test.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+var _ = Suite(&testScanSuite{})
+
+func (s *testScanSuite) SetUpSuite(c *C) {
+	var d Driver
+	store, err := d.Open(fmt.Sprintf("tikv://%s%s?cluster=%d", *etcdAddrs, *pdLeaderPath, *clusterID))
+	c.Assert(err, IsNil)
+	s.store = store.(*tikvStore)
+	s.prefix = "seek_cd7540"
```
Comment (disksing) [inline scan_test.go]: Maybe use formatted current timestamp? If you want tests' key ranges not overlap.

**Comment by disksing** [comment]
The rest LGTM.

File: scan_test.go (hunk 3 above)
Comment (morefreeze) [inline scan_test.go]: Seems better.

**Comment by disksing** [comment]
LGTM

**Comment by morefreeze** [comment]
PTAL

File: ticlient_test.go [hunk 4]
```diff
@@ -148,6 +153,12 @@ func (s *testTiclientSuite) TestCleanLock(c *C) {
...
 func (s *testTiclientSuite) TestNotExist(c *C) {
 	txn := s.beginTxn(c)
-	_, err := txn.Get(encodeKey(("noSuchKey")))
+	_, err := txn.Get(encodeKey(s.prefix, "noSuchKey"))
 	c.Assert(err, NotNil)
 }
+
+// s08d is for returning format string "%s%08d" to keep keys sorted.
+// e.g.: "0002" < "0011", otherwise "2" > "11"
+func s08d(prefix string, n int) string {
```
Comment (shenli) [inline ticlient_test.go]: I notice that you also use s08d to compose values: "err := txn.Set(k, []byte(s08d("value", i)))". So it is not only used on keys?

**Comment by coocood** [comment]
LGTM

**Pull request merged**